nuxeo-webengine/nuxeo-webengine-jaxrs/src/main/java/org/nuxeo/ecm/webengine/jaxrs/coreiodelegate/RenderingContextWebUtils.java

```diff
@@ -110,9 +110,11 @@ public static RenderingContextBuilder fillContext(RenderingContextBuilder builde
             HttpServletRequest webRequest = (HttpServletRequest) request;
             // base url
             String baseURL = VirtualHostHelper.getBaseURL(request);
-            // current session
-            CoreSession session = SessionFactory.getSession(webRequest);
-            builder.base(baseURL).session(session);
+            // current session if we have an authenticated user
+            if (webRequest.getUserPrincipal() != null) {
+                CoreSession session = SessionFactory.getSession(webRequest);
+                builder.base(baseURL).session(session);
```

Comment:
Should we fill the `baseURL` for non-authenticated user?

Reply:
When using an open URL, would NPE be thrown when the server is trying to close the session in somewhere else? Because there's no session to close.

Reply:
The server closes a session only if it created one. See `CoreSessionProvider`, `PerSessionCoreProvider`.

You're right for `baseURL`, will change that.